src/App.js

```diff
@@ -17,21 +18,29 @@ const makeRequest = async (endpoint, data) => {
 
 export function App() {
   const [synthetix, updateSynthetix] = useSynthetix();
-  const { walletAddress, connect, signature, signer } = synthetix;
+  const { walletAddress, connect, token, signer } = synthetix;
 
   const signupMutation = useMutation({
-    mutationFn: (data) => makeRequest('signup', data),
+    mutationFn: (data) => makeUnauthenticatedRequest('signup', data),
     onSuccess: ({ nonce }) =>
       signer.signMessage(nonce).then((signedMessage) => {
-        verificationMutation.mutate({ signedMessage });
+        verificationMutation.mutate({ nonce, signedMessage });
       }),
   });
 
   const verificationMutation = useMutation({
-    mutationFn: (data) => makeRequest('verify', data),
-    onSuccess: ({ signature }) => updateSynthetix({ signature }),
+    mutationFn: (data) => makeUnauthenticatedRequest('verify', data),
+    onSuccess: ({ token }) => {
+      saveToken({ walletAddress, token });
+      updateSynthetix({ token });
+    },
   });
 
+  const logout = () => {
+    window.localStorage.clear();
+    window.location.reload();
```

Comment:
```suggestion
    updateSynthetix(INITIAL_STATE);
```

need to reset the state instead of reloading page.

Reply:
made this change

Reply:
as discussed, you were right and we'd better reload the page. The only change for that then needs to be - `listAccounts` should not be called after reload after logout.